frontend/checks-runner.ts

```diff
@@ -0,0 +1,160 @@
+import checkbox, { Separator } from '@inquirer/checkbox';
+import chalk from 'chalk';
+import { spawn } from 'child_process';
+import fs from 'fs';
+import path from 'path';
+
+// Define the structure of package.json
+interface PackageJson {
+  scripts?: Record<string, string>;
+}
+
+// Define the allowed scripts with their descriptions and optional arguments
+type AllowedScripts = readonly {
+  value: string;
+  name: string;
+  description: string;
+  checked?: boolean;
+  seperator?: string;
+  args?: string;
+}[];
+
+// List of allowed scripts
+// prettier-ignore
+const allowedScripts: AllowedScripts = [
+  { value: 'format:check', name: 'Prettier', description: 'Opinionated code formatter', seperator: 'Quality Checks' },
+  { value: 'lint:check', name: 'ESLint', description: 'Statically analyzes your code to quickly find problems' },
+  { value: 'typecheck', name: 'Typescript', description: 'Run compiler by checking your code' },
+  { value: 'build', name: 'React Router', description: 'Build application and server', seperator: 'Build' },
+  { value: 'test', name: 'Vitest', description: 'Run next generation unit testing framework', args: 'run', seperator: 'Testing' },
+  { value: 'test:e2e', name: 'Playwright', description: 'Run reliable end-to-end testing for modern web apps' },
+];
+
+// Display welcome message
+console.log(chalk.green.bold('Welcome to the Checks Runner!\n'));
+console.log(chalk.magenta('This script allows you to select and run predefined npm scripts in sequence.\n'));
+
+// Keep track of current child process
+let currentChildProcess: ReturnType<typeof spawn> | null = null;
+
+// Handle parent process termination
+process.on('SIGINT', () => {
+  console.log(chalk.yellow('\nGracefully shutting down...'));
+  cleanup();
+  process.exit(0);
+});
+
+process.on('SIGTERM', () => {
+  console.log(chalk.yellow('\nGracefully shutting down...'));
+  cleanup();
+  process.exit(0);
+});
+
+// Cleanup function to kill child process
+function cleanup() {
+  if (currentChildProcess) {
+    console.log(chalk.yellow('Terminating running scripts...'));
+    currentChildProcess.kill('SIGTERM');
+    currentChildProcess = null;
+  }
+}
```

Comment:
If you name this function something more descriptive, there would be no need for a comment. For example:

``` ts
function terminateChildProcess()
```

Reply:
Code has been refactored.